addnewcitydialog.cpp

```diff
@@ -2,81 +2,68 @@
 #include "ui_addnewcitydialog.h"
 
 #include <QIcon>
+#include <QIntValidator>
+#include <QDoubleValidator>
 
 AddNewCityDialog::AddNewCityDialog(QWidget *parent) :
     QDialog(parent),
-    ui(new Ui::AddNewCityDialog),
-    m_messageBoxInfo{new QMessageBox(this)}
+    ui(new Ui::AddNewCityDialog)
 {
     ui->setupUi(this);
+
+    QIntValidator *intValidator = new QIntValidator(this);
+    intValidator->setBottom(0);
+    QDoubleValidator *doubleValidator = new QDoubleValidator(this);
+
+    ui->inputPopulationLineEdit->setValidator(intValidator);
+    ui->inputLowTempLineEdit->setValidator(doubleValidator);
+    ui->inputHighTempLineEdit->setValidator(doubleValidator);
+
+    connect(ui->saveCancelButtonBox, &QDialogButtonBox::rejected, this, &AddNewCityDialog::onSaveCancelButtonBoxRejected);
+    connect(ui->saveCancelButtonBox, &QDialogButtonBox::accepted, this, &AddNewCityDialog::onSaveCancelButtonBoxAccepted);
 }
 
 AddNewCityDialog::~AddNewCityDialog()
 {
     delete ui;
 }
 
-void AddNewCityDialog::on_save_cancel_buttonBox_accepted()
+void AddNewCityDialog::onSaveCancelButtonBoxAccepted()
 {
-    if (ui->inputContinentComboBox->currentText() == "Select continent"
-        || ui->inputCountryLineEdit->text().isEmpty()
-        || ui->inputCityLineEdit->text().isEmpty()
-        || ui->inputPopulationLineEdit->text().isEmpty()
-        || ui->inputHighTempLineEdit->text().isEmpty()
-        || ui->inputLowTempLineEdit->text().isEmpty())
-    {
-        m_messageBoxInfo->setIcon(QMessageBox::Critical);
-        m_messageBoxInfo->setText(QString("Cannot add new item"));
-        m_messageBoxInfo->setInformativeText(QString("Empty fields were found, please fill all data"));
-        m_messageBoxInfo->setDefaultButton(QMessageBox::Ok);
-        m_messageBoxInfo->show();
-        return;
-    }
-
-    bool wasConvertedPopulation(false), wasConvertedHighTemp(false), wasConvertedLowTemp(false);
-    //if (ui->inputCountryLineEdit->text().toUInt(&wasConverted); !wasConverted)
-    ui->inputPopulationLineEdit->text().toUInt(&wasConvertedPopulation);
-    ui->inputHighTempLineEdit->text().toInt(&wasConvertedHighTemp);
-    ui->inputLowTempLineEdit->text().toInt(&wasConvertedLowTemp);
+    auto continent = ui->inputContinentComboBox->currentText();
```

Comment:
inputContinentComboBox should be filled in with DataLoader::CONTINENTS. If there was no "Select continent" item you wouldn't have to check if user selected a value.